staging/src/k8s.io/client-go/tools/reference/ref.go

```diff
@@ -73,27 +55,32 @@ func GetReference(scheme *runtime.Scheme, obj runtime.Object) (*v1.ObjectReferen
 		listMeta = objectMeta
 	}
 
-	// if the object referenced is actually persisted, we can also get version from meta
+	gvk := obj.GetObjectKind().GroupVersionKind()
```

Comment:
I think we should get kind/version from obj.GetObjectKind() or from the scheme, not get them from different sources. e.g.:

```go
gvk := obj.GetObjectKind().GroupVersionKind()

if gvk.Empty() {
  gvks, _, err := scheme.ObjectKinds(obj)
  if err != nil {
    return nil, err
  }
  if len(gvks) == 0 || gvks[0].Empty() {
    return nil, fmt.Errorf("...")
  }
  gvk = gvks[0]
}

kind := gvk.Kind
version := gvk.GroupVersion().String()
```

Reply:
good point - done